### PR TITLE
Making input work

### PR DESCRIPTION
**Pull request by kitskub**
This is ready. If anyone thinks that this needs more stuff/changes, just comment.

SpoutAPI pull: https://github.com/SpoutDev/SpoutAPI/pull/334

### CODE REVIEW

File: src/main/java/org/spout/engine/SpoutEngine.java [hunk 1]
```diff
@@ -767,6 +769,10 @@ public Player addPlayer(String playerName, SpoutSession<?> session, int viewDist
 		while (true) {
 			player = onlinePlayers.getLive().get(playerName);
 
+			if (player == null) {
+				player = new SpoutPlayer(playerName, getDefaultWorld().getSpawnPoint(), session, this, viewDistance);
+			}
+
 			if (player != null) {
```
Comment (Zidane) [inline src/main/java/org/spout/engine/SpoutEngine.java]: Can't be null at this point

File: src/main/java/org/spout/engine/command/InputCommands.java [hunk 2]
```diff
@@ -45,32 +47,49 @@ public static void setupInputCommands(Engine engine, Command parent) {
...
 		}
 	}
 
 	public static class InputFlagHandler implements CommandExecutor {
 		private final PlayerInputState.Flags flag;
 		private final boolean add;
+		private final boolean client;
 
-		public InputFlagHandler(PlayerInputState.Flags flag, boolean add) {
+		public InputFlagHandler(PlayerInputState.Flags flag, boolean add, boolean client) {
```
Comment (Zidane) [inline src/main/java/org/spout/engine/command/InputCommands.java]: Bool is useless here...do a Spout.getPlatform check

File: src/main/java/org/spout/engine/SpoutEngine.java (hunk 1 above)
Comment (kitskub) [inline src/main/java/org/spout/engine/SpoutEngine.java]: Oh, that's true.

File: src/main/java/org/spout/engine/command/InputCommands.java (hunk 2 above)
Comment (kitskub) [inline src/main/java/org/spout/engine/command/InputCommands.java]: Good point

**Comment by Spouty** [comment]
**[AUTO-GENERATED COMMENT]**

Please make sure this pull request follows the [pull request guidelines](http://spout.in/prguide) in order to ensure your pull request is accepted.
- This pull request has commit(s) that are not signed off.
  
  Offending commit(s): 
- d22036a2f849b32b8a967663c400f9bebcc264cd

**Comment by macalinao** [comment]
:kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart: :kissing_heart:

**Comment by kitskub** [comment]
Comments?

**Comment by Salynne** [comment]
Looks good. Sorry I was a little busy this week, if you can get it auto-mergeable again I will pull them.

**Comment by kitskub** [comment]
It's not auto-mergeable anymore? Hmm, that should be a quick fix.

**Comment by kitskub** [comment]
It's auto-mergeable now.

**Pull request merged**